### PR TITLE
feat: big changes

### PR DESCRIPTION
**Pull request by borisdayma**
This PR is a mess but it's doing lots of things (it includes #1):

* [x] script for dataset in tfrecords/webp + dataloader
* [x] handle both rgb and lab space
* [x] fix quantize -> quantizer
* [x] handle multiple hosts (need infinite data in case we have different dataset length)
* [x] allow multiple configs: use of extra projection, extra conv post attention, bias, normformer, GLU, dropout
* [x] use both l1 (same as logit laplace?) and l2 loss
* [x] log metrics
* [x] support shampoo
* [x] use pjit for custom sharding
* [ ] log reconstruction samples
* [x] use lpips
* [ ] use stylegan loss
* [ ] use gradient penalty
* [ ] reset unused codes
* [ ] add scan layers

### CODE REVIEW

**Comment by borisdayma** [comment]
Current status:

* default LR
* 7h training

output

**Comment by borisdayma** [comment]
Pedro added LPIPS 🔥

**Comment by borisdayma** [comment]
All right, training loop now in working state

**Comment by pcuenca** [comment]
I can take a look at the stylegan loss to see what it looks like.

Regarding the `log reconstruction samples` task, should we do it using an external cron job like we usually do for training?

**Comment by borisdayma** [comment]
I'll add the remaining items as separate issues
